src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/ISyntaxFactsService.cs

```diff
@@ -32,6 +32,7 @@ internal interface ISyntaxFactsService : ILanguageService
         bool IsPredefinedOperator(SyntaxToken token);
         bool IsPredefinedOperator(SyntaxToken token, PredefinedOperator op);
         bool IsKeyword(SyntaxToken token);
+        bool IsReservedKeyword(string text);
```

Comment:
these helpers need documentation as to the difference between Keyword and ReservedKeyword.  examples woudl be good.